src/java/voldemort/client/protocol/admin/AdminClient.java

```diff
@@ -231,14 +232,17 @@ private AdminClient(AdminClientConfig adminClientConfig,
                                                                .getContextClassLoader());
 
         this.adminClientConfig = adminClientConfig;
+        this.clientConfig = clientConfig;
         this.socketPool = helperOps.createSocketPool(adminClientConfig);
         this.adminStoreClient = new AdminStoreClient(clientConfig);
         this.fetchSingleStore = !clientConfig.isFetchAllStoresXmlInBootstrap();
 
-        this.clusterVersion = System.currentTimeMillis();
         if(cluster != null) {
+            // We don't know at which time the Cluster's state was generated, so we can't know the cluster version.
+            this.clusterVersion = null;
             this.currentCluster = cluster;
         } else {
+            this.clusterVersion = System.currentTimeMillis();
```

Comment:
Recommend leaving a comment that time needs to be initalized before calling the bootstrap XML.